### ISSUE TITLE
[css-scrollbars] What is "the background"

### ISSUE THREAD

**Issue by frivoal** [w3c/csswg-drafts#10524]
In https://github.com/w3c/csswg-drafts/issues/9853, we resolved that:
> transparent colors in the thumb are transparent towards the track, and transparent colors in the track are transparent towards the background […]

But what exactly is "the background"? As far as I can tell from testing, Firefox and Chrome do agree: this is the composited background of all elements that end up behind, including images, gradients… and if the backgrounds themselves are transparent, you include further backgrounds behind. It does not include the content of the scrolling element, but it does include the content of ancestors / things behind:

http://software.hixie.ch/utilities/js/live-dom-viewer/?saved=12876

<img width="257" alt="Screenshot 2024-07-04 at 10 20 37" src="https://github.com/w3c/csswg-drafts/assets/113268/b32c6d35-4db2-4000-ade0-7e50142a09b8">

Since implementations agree, I see no strong reason to propose anything else, but is there a pre-existing term to refer to this kind of background?

I'm guessing that it is effectively transparent towards everything, not just things that are technically background, and it is merely that the content of the scrolling element itself is clipped (to the padding box), so there's nothing to be transparent to.

**Comment by astearns** [comment]
@frivoal since this will be attempting to specify reality, could we take this as an async resolution? Perhaps if you had a proposed resolution, people would reply with corrections 😁

**Comment by flackr** [comment]
> I'm guessing that it is effectively transparent towards everything, not just things that are technically background, and it is merely that the content of the scrolling element itself is clipped (to the padding box), so there's nothing to be transparent to.

Yes, that's my understanding of what is happening in Chrome at least. Note that if you set background-attachment: local on the scrolling container the background is also clipped to the padding box: https://software.hixie.ch/utilities/js/live-dom-viewer/?saved=13412

Effectively the non-overlay scrollbars (and non-local background) extend beyond the scrolling content. Overlay scrollbars draw overtop of the scrolling content.

Note that we have talked in other issues about changing the scrolling content clip - so I think it would be more correct to just say that the thumb is composited above the track which is composited above the scrolling container (and scrolling contents if they are not clipped or if using overlay scrollbars).

**Comment by kbabbitt** [comment]
> the thumb is composited above the track which is composited above the scrolling container (and scrolling contents if they are not clipped or if using overlay scrollbars)

...and the scrolling container is, as usual, composited above content that's behind it in z-order? Makes sense to me.

**Comment by frivoal** [comment]
> Note that if you set background-attachment: local on the scrolling container the background is also clipped to the padding box: https://software.hixie.ch/utilities/js/live-dom-viewer/?saved=13412

That specific part is true in chrome, but not in Firefox. Is that already observable in different circumstances, or covered by some spec, or would we be here looking to define which one is correct?

**Comment by emilio** [comment]
https://searchfox.org/mozilla-central/rev/0b189f017bc9d48b62012205c5b9f8a8b560497b/layout/generic/ScrollContainerFrame.cpp#3854,3873 fwiw

**Comment by emilio** [comment]
Which comes explicitly from https://bugzilla.mozilla.org/show_bug.cgi?id=636564, so that's gonna be fun
